src/components/annotations/draw.js

```diff
@@ -62,47 +63,65 @@ function draw(gd) {
  * index (int): the annotation to draw
  */
 function drawOne(gd, index) {
-    var layout = gd.layout,
-        fullLayout = gd._fullLayout,
-        gs = gd._fullLayout._size;
+    var fullLayout = gd._fullLayout;
+    var options = fullLayout.annotations[index] || {};
+    var xa = Axes.getFromId(gd, options.xref);
+    var ya = Axes.getFromId(gd, options.yref);
 
-    // remove the existing annotation if there is one
-    fullLayout._infolayer.selectAll('.annotation[data-index="' + index + '"]').remove();
+    drawRaw(gd, options, index, xa, ya);
+}
 
-    // remember a few things about what was already there,
-    var optionsIn = (layout.annotations || [])[index],
-        options = fullLayout.annotations[index];
+/*
+ * drawRaw: draw a single annotation, potentially with modifications
+ *
+ * options (object): this annotation's options
+ * index (int): the annotation to draw
+ * xa (object | undefined): full x-axis object to compute subplot pos-to-px
+ * ya (object | undefined): ... y-axis
+ */
+function drawRaw(gd, options, index, xa, ya) {
+    var fullLayout = gd._fullLayout;
+    var gs = gd._fullLayout._size;
+
+    var className = options._scene ?
```

Comment:
elsewhere `_scene` is an object, but this one is an id - can we call it `_sceneId` instead?